apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx

```diff
@@ -133,26 +141,103 @@ const getNativeElementsList = (elements: SBEnumType['value']): JSX.Element => {
   );
 };
 
-const RenderArgsTable = ({ hideArgsTable, primaryStory }: { primaryStory: PrimaryStory; hideArgsTable: boolean }) => {
+const slotRegex = /as\?:\s*"([^"]+)"/;
+let hasSlotMatch = false;
+function withSlotEnhancer(story: PreparedStory) {
+  const updatedArgTypes = { ...story.argTypes };
+
+  type InternalComponentApi = {
+    __docgenInfo: { props?: Record<string, { type: { name: string } }> };
+    [k: string]: unknown;
+  };
+  const component = story.component as InternalComponentApi;
+  const docGenProps = component?.__docgenInfo?.props;
+
+  if (!docGenProps) {
+    return component;
+  }
+
+  Object.entries(docGenProps).forEach(([key, argType]) => {
+    const value: string = argType?.type?.name;
+    if (value.includes('WithSlotShorthandValue')) {
+      hasSlotMatch = true;
+      const match = value.match(slotRegex);
+      if (match) {
+        component.__docgenInfo.props![key].type.name = `Slot<\"${match[1]}\">`;
+        // @ts-expect-error - storybook doesn't ship proper types (value is missing)
+        updatedArgTypes[key].type.value = `Slot<\"${match[1]}\">`;
+      } else {
+        component.__docgenInfo.props![key].type.name = `Slot`;
+        // @ts-expect-error - storybook doesn't ship proper types (value is missing)
+        updatedArgTypes[key].type.value = `Slot`;
+      }
+    }
+  });
+
+  return component;
+}
+
+const AdditionalApiDocs: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
AFAIR this is redundant generic instantiation 

<img width="798" alt="image" src="https://github.com/user-attachments/assets/52a69081-b790-45b6-8a43-3271d941a430" />


```suggestion
const AdditionalApiDocs: React.FC= ({ children }) => {
```

Reply:
nice this is a good change. @carlamntn lets keep it as is but change the ReactNode for ReactElement. Reactnode is anti-pattern as it unwraps to `any` more or less